fe_utils/mesh.py

```diff
@@ -111,7 +112,12 @@ def jacobian(self, c):
         :result: The Jacobian for cell ``c``.
         """
 
-        raise NotImplementedError
+        cg1 = LagrangeElement(self.cell, 1)
+        cg1fs = FunctionSpace(self, cg1)
```

Comment:
If `n` is mesh size then global numbering is `O(n)`, so this makes the cost of evaluating one cell Jacobian `O(n)` and the cost of evaluating an integral over a mesh `O(n^2)`, which is very sad. You need to lift the things that don't depend on `c` up into the mesh constructor.

Reply:
makes sense, think i've fixed this now